### PR TITLE
View comments, dialog box and image preview

### PR DESCRIPTION
**Pull request by kshitij01042002**
This pr consist of two issues addressed
#108 #62 
image

### CODE REVIEW

**Comment by narayan954** [comment]
> Since I worked continuously, I resolved three issues and thus made a single PR for it

That's good Kshitij, but the proper developer practice is creating separate branches for different features. Currently, branch view-comments does multiple works and even some of them out of its domain like dialog box and image preview. 
For now, I can merge them but please follow proper practices, they're good in the long run.

**Comment by kshitij01042002** [comment]
Okay! Works. This being my first lesson towards version controlling 
Thank You!!!

**Review by narayan954 — COMMENTED** [review]
You shouldn't have deleted the firebase.js file.

**Comment by narayan954** [comment]
please update your current branch with the master and restore firebase.js file.

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@ dist-ssr
 # firebase
 /.firebase
 /firebase
+/lib/firebase.js
```
Comment (narayan954) [inline .gitignore:31]: this shouldn't be in the gitignore as this file has stuffs to be changed by others!

File: src/components/Post.jsx [hunk 2]
```diff
@@ -299,21 +309,28 @@ function Post(prop) {
...
-                {user && userComment.content.username === user.displayName ? (
-                  <DeleteTwoToneIcon fontSize="small" />
-                ) : (
-                  <></>
-                )}
-              </span>
-            </p>
-          ))}
-        </div>
+        <Button onClick={setisCommentOpen}>View All comments</Button>
```
Comment (narayan954) [inline src/components/Post.jsx:312]: check if comments exist, then display view all comments

Comment (kshitij01042002) [reply src/components/Post.jsx]: Done!

**Comment by kshitij01042002** [comment]
Merge my this pull request under codepeak @narayan954

**Comment by narayan954** [comment]
> Merge my this pull request under codepeak @narayan954

but this would be counted as 1 in codepeak through automated process

File: src/components/Post.jsx [hunk 3]
```diff
@@ -298,22 +308,38 @@ function Post(prop) {
...
-                  <DeleteTwoToneIcon fontSize="small" />
-                ) : (
-                  <></>
-                )}
-              </span>
-            </p>
-          ))}
-        </div>
+        
+        {comments.length != 0 ? (
```
[the remark this answers is not in the document]
Comment (kshitij01042002) [reply src/components/Post.jsx]: Done

**Pull request merged**